minio/api.py

```diff
@@ -1384,21 +1389,29 @@ def presigned_post_policy(self, post_policy):
         credential_string = generate_credential_string(self._access_key,
                                                        date, region)
 
-        post_policy_base64 = post_policy.base64(extras=[
+        policy = [
             ('eq', '$x-amz-date', iso8601_date),
             ('eq', '$x-amz-algorithm', _SIGN_V4_ALGORITHM),
             ('eq', '$x-amz-credential', credential_string),
-        ])
+        ]
+        if self._session_token:
+            policy.add(('eq', '$x-amz-security-token', self._session_token))
+
+        post_policy_base64 = post_policy.base64(extras=policy)
         signature = post_presign_signature(date, region,
                                            self._secret_key,
                                            post_policy_base64)
-        post_policy.form_data.update({
+        form_data = {
             'policy': post_policy_base64,
             'x-amz-algorithm': _SIGN_V4_ALGORITHM,
             'x-amz-credential': credential_string,
             'x-amz-date': iso8601_date,
             'x-amz-signature': signature,
-        })
+        }
+        if self._session_token:
```

Comment:
I need to do that before such that signature can be generated for policy, also the flow is fine since I didn't want to move the code around too much to look weird.